src/facade/dragonfly_connection.cc

```diff
@@ -1098,16 +1098,18 @@ void Connection::DispatchSingle(bool has_more, absl::FunctionRef<void()> invoke_
   bool optimize_for_async = has_more;
   QueueBackpressure& qbp = GetQueueBackpressure();
   if (optimize_for_async &&
-      qbp.IsPipelineBufferOverLimit(stats_->dispatch_queue_bytes, dispatch_q_.size())) {
+      qbp.IsPipelineBufferOverLimit(stats_->dispatch_queue_bytes, stats_->dispatch_queue_entries)) {
```

Comment:
dispatch_queue_entries is global per io thread
moving to use it changes the behaviour

ABSL_FLAG(uint32_t, pipeline_queue_limit, 10000,
          "Pipeline queue max length, the server will stop reading from the client socket"
          " once its pipeline queue crosses this limit, and will resume once it processes "

Now you allow only 10000 pipline messages for all connections

Reply:
Current change will allow only 10000 (by default) pipeline message for all connections. I'll revert back to use dispatch_q for each connection.